code/modules/client/preferences.dm

```diff
@@ -1480,57 +1464,57 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 						be_special += be_special_type
 
 				if("rand_gender")
-					random_gender = !random_gender
+					randomise["gender"] = !randomise["gender"]

```

Comment:
You can simplify all this by passing the randomization type in topic with something like this:
```
in html:
<a href='?_src_=prefs;preference=random_toggle;random_type=[RANDOM_GENDER]'>clickme</a>

and in here:
if("rand_toggle")
  var/rand_type = href_list["random_type"]
  //Check if it's actual value here
  randomise[rand_type] = !randomise[rand_type]
```